cli/internal/terraform/terraform/azure/modules/scale_set/main.tf

```diff
@@ -11,6 +11,29 @@ terraform {
   }
 }
 
+locals {
+  role_dashed = var.role == "ControlPlane" ? "control-plane" : "worker"
+  tags = merge(
+    var.tags,
+    { constellation-role = local.role_dashed },
+    { constellation-node-group = var.node_group_name },
+  )
+  # migration: allow the old node group names to work since they were created without the uid
```

Comment:
The migration probably has a start version (from which version on do we need this migration/do we start migrating)  and an end version (when have we migrated / can expect the current infrastructure to fit the new assumptions). Then we can more easily remove this migration when it is no longer needed.
Please add that information here or in other parts of the migration and refer to it by e.g. referring to the same ticket.

Reply:
the distinction of the current naming will only become obsolete once all cluster with a version <v2.9 are destroyed. Not sure at what point we want users to force to switch. It was also done like this in the GCP PR.
The existing name cannot be changed because it would force a recreate. 
Any suggestion?

Reply:
Hm, then maybe the name "migration" is a bit misleading an "compatibility" is more correct? Since we want to upload compatibility with the old version and don't change (aka. migrate) any resources, as you mentioned.

@malt3 What do you think of this? 

How long do we expect this piece of code to be there? I'd want to get rid of it _eventually_ (in 3-4 versions?). We can always have migrations with downtime, as long as they can be prepared for.

Reply:
afaik we can't ever remove this backwards compatibility option because it will break upgrade compatibility with any version of Constellation created before the version this features is released in.

Reply:
I thought about this naming change for some time now.
The whole migration / backwards compatibility for the scale set name can be skipped (and I can also remove it from the gcp template), as long as we get the lifecycle block correct.
This means:

- Existing clusters can have scale sets with the legacy names, they will not be replaced because of the lifecycle ignore annotations
- New clusters will use the new naming scheme even for the two default scale sets

Reply:
so we remove the move block and add the old resources to ignore_changes?

Reply:
I'm wondering if the benefit of not having the migration name check outweighs that we are then unable to change the legacy resource at all

Reply:
No. The move block stays. This is about the following naming thing:

```
  group_uid = random_id.uid.hex
  maybe_uid = (var.node_group_name == "control_plane_default" || var.node_group_name == "worker_default") ? "" : "-${local.group_uid}"
  #maybe_one = var.node_group_name == "worker_default" ? "-1" : "" # TODO adding "-1" forces replaces of resource on upgrade
  name = "${var.base_name}-${local.role_dashed}${local.maybe_uid}"
```

And I would like to simplify it to always use the new naming scheme:

```
  group_uid = random_id.uid.hex
  name = "${var.base_name}-${local.role_dashed}${local.group_uid}"
```

Additionally, you need to explicitly ignore changes of the name for the scale set in the lifecycle block:

```hcl
resource "azurerm_linux_virtual_machine_scale_set" "scale_set" {

    # removed for brevity [...]

  lifecycle {
    ignore_changes = [
      name, # required. Allow legacy scale sets to keep their old names
      instances,       # required. autoscaling modifies the instance count externally
      source_image_id, # required. update procedure modifies the image id externally
    ]
  }

}
```

Reply:
gotcha, thanks!